Colony match updated and gradle properties

Aaron9174 opened the pull request:
Colony match needs to take dimension ID for context. Colony IDs in different dimensions start at 1, so if you have two colonies in different dimensions trying to be allies, they won't show up in the list since the colony ID is not unique. Colony ID + dimension ID is unique and that's what this change involves.

Also, gradle properties setup to run the required version of gradle. Perhaps I have misconstrued the README directions, but I couldn't get this to run with JVM21, and instead got this working with JVM17 + a different gradle version. Possible that the gradle version change can be removed and only the ColonyView change is needed. Has been tested locally and on a server on version 1.20.1 at the most recent tag.

Closes #
Closes #

# Changes proposed in this pull request
- Updating colony ID check for populating ally list to include dimension ID as well
- Gradle version for build distribution

## Testing
- [ x ] Yes I tested this before submitting it.
- [ x ] I also did a multiplayer test.

Review please

Aaron9174 committed:
Colony match updated and gradle properties

Colony match needs to take dimension ID for context. Also, gradle
properties setup to run the required version of gradle.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/ldtteam/minecolonies?pullRequest=10928) <br/>All committers have signed the CLA.

Aaron9174 committed:
Merge branch 'version/main' into bug/allies-list-empty-on-diff-dimensions

Nightenom commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
```
Whenever you change version of gradle wrapper please run twice `gradlew wrapper --gradle-version [version]`

Aaron9174 replied:
Ah that makes sense, I uh, definitely just wrote this in by hand which isn't the correct way to do it lol
edit: All done

uecasm commented:
JVM21 is only for the 1.21 branch.  The main branch is still 1.20 with JVM17.

Aaron9174 commented:
> JVM21 is only for the 1.21 branch. The main branch is still 1.20 with JVM17.

This should be updated in the 1.20.1 tagged README then. I will update the README for this branch, so in the next tagged 1.20.1 patch update it will be more accurate.

Aaron9174 committed:
Updating gradle version

Updating gradle version properly (use gradlew wrapper instead of hard
coding the version directly in the properties file)

Aaron9174 committed:
Merge branch 'bug/allies-list-empty-on-diff-dimensions' of github.com:Aaron9174/minecolonies into bug/allies-list-empty-on-diff-dimensions

Aaron9174 committed:
Updated README to match the version needed by the 1.20.1 tag.

Aaron9174 committed:
Gradle version updated to match main

Aaron9174 committed:
Merge branch 'version/main' of github.com:ldtteam/minecolonies into bug/allies-list-empty-on-diff-dimensions

Aaron9174 commented:
All comments and merge conflicts addressed, looking for a re-review here

Raycoms commented on src/main/java/com/minecolonies/core/colony/ColonyView.java:
```diff
@@ -377,7 +377,8 @@ public static void serializeNetworkData(@NotNull Colony colony, @NotNull Friendl
                 for (final ColonyPlayer owner : colony.getPermissions().getPlayersByRank(colony.getPermissions().getRankOwner()))
                 {
-                    if (col.getPermissions().getRank(owner.getID()).isColonyManager() && col.getID() != colony.getID())
+                    boolean colonyMatch = (col.getID() == colony.getID()) && (col.getDimension() == colony.getDimension());
```
I would probably inline this as && (id != id || dim != dim), no?

Aaron9174 replied:
You got it chief, don't mind making the adjustment

Aaron9174 replied:
The change is in, but after re-visiting this, I think it was better before I made the change. As it was more readable via a variable name indicating there was a colonyMatch. Further, the boolean algebra was just simpler to understand before, even with context of these changes, I still had to whip out a truth table to make sure this logic was identical (which it is).

I don't readily see from this statement
col.getID() != colony.getID() || col.getDimID() != colony.getDimID() 
that the current colony of the loop doesn't match the ID of the colony sent in the network request.

Aaron9174 committed:
Made change inline

Truth table confirmed the logic is the same

Aaron9174 commented:
@Nightenom Got Raycoms approval, mind taking a look at this when you get a chance?

Aaron9174 commented:
Was building locally wrong, just using the ./gradlew (without build), was just missing an parenthesis and was confused why it worked locally. Committing now.

Aaron9174 committed:
Fixed parenthesis error

Missing a closing parenthesis
